tests/unit/test_gate_requirements.py

```
@@ -506,3 +523,112 @@ def test_every_declared_requirement_is_well_formed(self):
+        from slopmop.doctor.gate_preflight import _missing_required_tools
+
+        class _OptionalToolGate(_ToollessGate):
+            def requirements(self) -> Requirements:
+                return Requirements(
+                    items=(
+                        Requirement(kind="system", name="opt", optional=True),
+                        Requirement(kind="system", name="req", optional=False),
+                    )
+                )
+
+        monkeypatch.setattr("slopmop.checks.base.find_tool", lambda n, r: None)
```

Ruff (0.15.17)

[warning] Unused lambda argument: `n`

(ARG005)